src/mc.rs

```diff
@@ -1230,3 +1217,1063 @@ pub unsafe fn resize_rust<BD: BitDepth>(
+            tmp1: *const int16_t,
+            tmp2: *const int16_t,
+            w: libc::c_int,
+            h: libc::c_int,
+            mask: *const uint8_t,
+            bitdepth_max: libc::c_int,
+        );
+    };
+
+    (w_mask, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            tmp1: *const int16_t,
+            tmp2: *const int16_t,
+            w: libc::c_int,
+            h: libc::c_int,
+            mask: *mut uint8_t,
+            sign: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+
+    (blend, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            tmp: *const DynPixel,
+            w: libc::c_int,
+            h: libc::c_int,
+            mask: *const uint8_t,
+        );
+    };
+
+    (blend_dir, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            tmp: *const DynPixel,
+            w: libc::c_int,
+            h: libc::c_int,
+        );
+    };
+
+    (warp8x8, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            src: *const DynPixel,
+            src_stride: ptrdiff_t,
+            abcd: *const int16_t,
+            mx: libc::c_int,
+            my: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+
+    (warp8x8t, $name:ident) => {
+        pub(crate) fn $name(
+            tmp: *mut int16_t,
+            tmp_stride: ptrdiff_t,
+            src: *const DynPixel,
+            src_stride: ptrdiff_t,
+            abcd: *const int16_t,
+            mx: libc::c_int,
+            my: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+
+    (emu_edge, $name:ident) => {
+        pub(crate) fn $name(
+            bw: intptr_t,
+            bh: intptr_t,
+            iw: intptr_t,
+            ih: intptr_t,
+            x: intptr_t,
+            y: intptr_t,
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            src: *const DynPixel,
+            src_stride: ptrdiff_t,
+        );
+    };
+
+    (resize, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            src: *const DynPixel,
+            src_stride: ptrdiff_t,
+            dst_w: libc::c_int,
+            h: libc::c_int,
+            src_w: libc::c_int,
+            dx: libc::c_int,
+            mx: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+}
+
+#[cfg(feature = "asm")]
+macro_rules! decl_fns {
+    ($ty:ident, $name:ident) => {
+        decl_fns!($ty, $name, sse2);
+        decl_fns!($ty, $name, ssse3);
+        decl_fns!($ty, $name, avx2);
+        decl_fns!($ty, $name, avx512icl);
+    };
+
+    ($ty:ident, $name:ident, $suffix:ident) => {
```

Comment:
I think `$ty` is kind of confusing since `ty` usually means an actual type.  Maybe `$mc_kind`?

Reply:
Changed.